src/EnergyPlus/EPVector.hh

```diff
@@ -98,20 +99,18 @@ template <typename T> struct EPVector : private std::vector<T>
 
     void allocate(std::size_t size)
     {
-        m_allocated = true;
         this->resize(size);
         std::fill(begin(), end(), T{});
     }
 
     void deallocate() noexcept
     {
-        m_allocated = false;
         this->clear();
     }
 
     [[nodiscard]] bool allocated() const noexcept
     {
-        return m_allocated || !this->empty();
```

Comment:
Using `||` here means that calling `clear` will make the vector empty but this function will still return true.